src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,22 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleConfirm }) {
+    super({ popupSelector });
+    this._handleConfirm = handleConfirm;
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__button_delete-card"
+    );
+    this.setEventListeners();
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._confirmButton.addEventListener("click", (event) => {
+      event.preventDefault();
+      this._handleConfirm();
+      this.setButtonText(".modal__button_delete-card", "Saving...");
```

Comment:
You have found the button as `this._confirmButton`. No need to search it again inside methods